source/extensions/quic_listeners/quiche/active_quic_listener.cc

```diff
@@ -73,11 +72,14 @@ ActiveQuicListener::ActiveQuicListener(Event::Dispatcher& dispatcher,
       listener_config.udpPacketWriterFactory()->get().createUdpPacketWriter(
           listen_socket_.ioHandle(), listener_config.listenerScope());
   udp_packet_writer_ = udp_packet_writer.get();
-  if (udp_packet_writer->isBatchMode()) {
-    // UdpPacketWriter* can be downcasted to UdpGsoBatchWriter*, which indirectly inherits
-    // from the quic::QuicPacketWriter class and can be passed to InitializeWithWriter().
-    quic_dispatcher_->InitializeWithWriter(
-        dynamic_cast<Quic::UdpGsoBatchWriter*>(udp_packet_writer.release()));
```

Comment:
I removed the reference to a now linux-specific type from this code. Also, it seemed possible that a non-quic impl could return true from `isBatchMode` so I refactored the check. If we think that's safe, might as well make it a `static_cast`, possibly with a debug-assert that the cast succeeds.

Reply:
Brilliant idea! static_cast is not safe if in the future other batch writer implementations are there.